OnTheMapTest/OTMLoginViewController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  ViewController.swift
+//  OnTheMapTest
+//
+//  Created by Gareth O'Sullivan on 17/12/2018.
+//  Copyright © 2018 Locust Redemption. All rights reserved.
+//
+
+import UIKit
+
+class OTMLoginViewController: UIViewController {
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
+    @IBAction func Login() {
+        var request = URLRequest(url: URL(string: "https://onthemap-api.udacity.com/v1/session")!)
+        request.httpMethod = "POST"
+        request.addValue("application/json", forHTTPHeaderField: "Accept")
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.httpBody = "{\"udacity\": {\"username\": \"\(emailTextField.text!)\", \"password\": \"\(passwordTextField.text!)\"}}".data(using: .utf8)
+        let session = URLSession.shared
+        let task = session.dataTask(with: request) { data, response, error in
+            guard error == nil else {
+                //handle error
+                return
+            }
+            // Add check for valid status code
+            // Add check to ensure result receive
+            let range = Range(5..<data!.count)
+            let result = data?.subdata(in: range)
+            let parsedResult: [String:AnyObject]!
```

Comment:
You could move all of this from the start of the function down to here into a separate class i.e. `UdacityLogin`. You could have a function called login which takes username and password as parameters

Reply:
Yes, the final project will have a number of files with separate classes to store constants, functions etc.

I am just getting the code working and functional. I know this might be longer but I will focus on the layout and structure in the final project.